ui/app/adapters/watchable.js

```diff
@@ -111,33 +111,36 @@ export default class Watchable extends ApplicationAdapter {
     return this.ajax(urlPath, 'GET', {
       signal,
       data: params,
-    }).then((payload) => {
-      const adapter = store.adapterFor(type.modelName);
-
-      // Query params may not necessarily map one-to-one to attribute names.
-      // Adapters are responsible for declaring param mappings.
-      const queryParamsToAttrs = Object.keys(
-        adapter.queryParamsToAttrs || {}
-      ).map((key) => ({
-        queryParam: key,
-        attr: adapter.queryParamsToAttrs[key],
-      }));
-
-      // Remove existing records that match this query. This way if server-side
-      // deletes have occurred, the store won't have stale records.
-      store
-        .peekAll(type.modelName)
-        .filter((record) =>
-          queryParamsToAttrs.some(
-            (mapping) => get(record, mapping.attr) === query[mapping.queryParam]
-          )
-        )
-        .forEach((record) => {
+    })
+      .then((payload) => {
+        const adapter = store.adapterFor(type.modelName);
+
+        // Query params may not necessarily map one-to-one to attribute names.
+        // Adapters are responsible for declaring param mappings.
+        const queryParamsToAttrs = Object.keys(
+          adapter.queryParamsToAttrs || {}
+        ).map((key) => ({
+          queryParam: key,
+          attr: adapter.queryParamsToAttrs[key],
+        }));
+
+        // Remove existing records that match this query. This way if server-side
+        // deletes have occurred, the store won't have stale records.
+        const matchingRecords = store.peekAll(type.modelName).filter((record) =>
+          queryParamsToAttrs.some((mapping) => {
+            if (mapping.queryParam === 'namespace' && query.namespace === '*')
+              return true;
+            return get(record, mapping.attr) === query[mapping.queryParam];
```

Comment:
@philrenaud This algorithm needs more work. I tried to make it minimal change by only adding one life, which works great if no filters are applied... however, if we apply the `namespace` filter, we're could run into issues.